editor/editor_node.cpp

```diff
@@ -4783,34 +4783,38 @@ void EditorNode::_pick_main_scene_custom_action(const String &p_custom_action_na
 	}
 }
 
-Ref<Texture2D> EditorNode::_get_class_or_script_icon(const String &p_class, const Ref<Script> &p_script, const String &p_fallback, bool p_fallback_script_to_theme) {
-	ERR_FAIL_COND_V_MSG(p_class.is_empty(), nullptr, "Class name cannot be empty.");
+Ref<Texture2D> EditorNode::_get_script_icon(const Ref<Script> &p_script, bool p_fallback_script_to_theme) {
+	ERR_FAIL_COND_V_MSG(p_script.is_null(), nullptr, "Script should be valid.");
 	EditorData &ed = EditorNode::get_editor_data();
 
-	// Check for a script icon first.
-	if (p_script.is_valid()) {
-		Ref<Texture2D> script_icon = ed.get_script_icon(p_script);
-		if (script_icon.is_valid()) {
-			return script_icon;
-		}
+	Ref<Texture2D> script_icon = ed.get_script_icon(p_script);
+	if (script_icon.is_valid()) {
+		return script_icon;
+	}
 
-		if (p_fallback_script_to_theme) {
-			// Look for the native base type in the editor theme. This is relevant for
-			// scripts extending other scripts and for built-in classes.
-			String script_class_name = p_script->get_language()->get_global_class_name(p_script->get_path());
-			String base_type;
-			if (script_class_name.is_empty()) {
-				base_type = p_script->get_instance_base_type();
-			} else {
-				base_type = ScriptServer::get_global_class_native_base(script_class_name);
-			}
+	if (p_fallback_script_to_theme) {
+		// Look for the native base type in the editor theme. This is relevant for
+		// scripts extending other scripts and for built-in classes.
+		String script_class_name = p_script->get_language()->get_global_class_name(p_script->get_path());
+		String base_type;
+		if (script_class_name.is_empty()) {
+			base_type = p_script->get_instance_base_type();
+		} else {
+			base_type = ScriptServer::get_global_class_native_base(script_class_name);
+		}
 
-			if (theme.is_valid() && theme->has_icon(base_type, EditorStringName(EditorIcons))) {
-				return theme->get_icon(base_type, EditorStringName(EditorIcons));
-			}
+		if (theme.is_valid() && theme->has_icon(base_type, EditorStringName(EditorIcons))) {
+			return theme->get_icon(base_type, EditorStringName(EditorIcons));
 		}
 	}
 
+	return nullptr;
+}
+
+Ref<Texture2D> EditorNode::_get_class_icon(const String &p_class, const String &p_fallback) {
+	ERR_FAIL_COND_V_MSG(p_class.is_empty(), nullptr, "Class name cannot be empty.");
+	EditorData &ed = EditorNode::get_editor_data();
```

Comment:
```suggestion
	EditorData &ed = get_editor_data();
```
Although the variable feels redundant.